libcaf_core/caf/actor_system_config.hpp

```diff
@@ -93,9 +94,19 @@ class CAF_CORE_EXPORT actor_system_config {
   /// @experimental
   template <class T, class... Ts>
   actor_system_config& add_actor_type(std::string name) {
```

Comment:
In the end, we always call `add_actor_factory(std::move(name), make_actor_factory<T, Ts...>());`.

We could just always look at the result from `make_behavior` and deduce the actor handle from that. That would get around the `if constexpr` logic.

Reply:
This could also be a way to evade including `actor_from_state` here.

Reply:
With the new refactoring, `T` picks up the whole `actor_from_state<S>` so there is no need for the header.